src/carousel/index.ts

```diff
@@ -0,0 +1,283 @@
+    const speed = this.props.settings.speed || DEFAULT_SETTINGS.speed;
+
+    const trackWidth = (slideCount + 2 * slidesToShow) * slideWidth;
+    const pos = this.calcPos(this.currentSlide, slideWidth);
+    const tfm = `translate3d(-${pos}px, 0px, 0px)`;
+
+    const transformStyles = {
+      transform: tfm,
+      '-webkit-transform': tfm,
+      '-ms-transform': tfm,
+    };
+
+    const tsVal = speed + 'ms ' + 'ease';
+    const transitionStyles = transition === true ? {
+      '-webkit-transition': tsVal,
+      transition: tsVal
+    } : {};
+
+    const style = Object.assign({}, {
+      width: `${trackWidth}px`,
+    },
+      transformStyles,
+      transitionStyles);
+
+    if (this.temporaryNoTransition) {
+      delete style['transition'];
+      delete style['-webkit-transition'];
+      this.temporaryNoTransition = !this.temporaryNoTransition;
+    }
+    return style;
+  }
+
+  getDots = () => {
+    if (!this.props.items) {
+      return;
+    }
+    const slideCount = this.props.items.length;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    const dotCount = Math.ceil((slideCount - slidesToShow) / slidesToScroll + 1);
+
+    return Array(dotCount).fill('dot');
+  }
+
+  getDotStyle = (i: number) => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    let leftBound = i * slidesToScroll;
+    let rightBound = i * slidesToScroll + slidesToShow - 1;
+
+    if (this.currentSlide >= leftBound && this.currentSlide <= rightBound) {
+      return { 'background-color': 'black' };
+    } else {
+      return {};
+    }
+  }
+
+  swipeSlides = (touchObj: { startX: number, startY: number, curX: number, curY: number }) => {
+    const direction: string = calSwipeDirection(touchObj);
+
+    if (direction === 'left') {
+      this.moveNext();
+    } else if (direction === 'right') {
+      this.movePrevious();
+    }
+  }
+
+  getSlideWidth = () => {
+    const visibleWidth = this.refs.carouselwrap.offsetWidth;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    if (visibleWidth) {
+      const slideWidth = visibleWidth / slidesToShow;
+      return slideWidth;
+    }
+  }
+
+  calcPos = (currS: number, moveDistance: number): number => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    return (currS + slidesToShow) * moveDistance;
+  }
+}
+
+const calSwipeDirection = (touchObj: { startX: number, startY: number, curX: number, curY: number }): string => {
+  const xDist = touchObj.startX - touchObj.curX;
+  const yDist = touchObj.startY - touchObj.curY;
+  const r = Math.atan2(yDist, xDist);
```

Comment:
Is it necessary to calculate this? Can't you just use the start x and the end x and see the difference?